qiskit/visualization/matplotlib.py

```diff
@@ -756,10 +756,13 @@ def _draw_ops(self, verbose=False):
 
                 # add conditional
                 if op.condition:
-                    cond_xy = [
-                        self._c_anchors[ii].plot_coord(this_anc, layer_width, self._x_offset)
-                        for ii in self._clbits_dict
-                    ]
+                    cond_xy = []
+                    for clbit in self._clbits_dict:
+                        xpos = self._data[node]["q_xy"][0][0]
+                        ypos = self._clbits_dict[clbit]["y"] - ((this_anc + 1) // self._fold) * (
+                            n_lines + 1
+                        )
```

Comment:
Entirely minor, but `xpos` can be calculated outside the loop, and the inner loop can be
```python
for clbit_dict in self._clbits_dict.values():
    ypos = clbit_dict["y"] - ...
```
just to make things a little cleaner.

On a larger note, though, why is it not appropriate to use `Anchor.plot_coord` in this case, like while plotting qubits?  My cursory reading of the code feels like logically this should be a job for that function, and if it's appropriate, it'd be nicer to fix any bug in there than to avoid calling it.

Reply:
@jakelishman So "after further consideration", it probably makes sense to use `plot_coord`. The Anchor code has some issues with naming and I took this opportunity to clean that up. For example `this_anc` is not an Anchor, but an `x_index`, so I changed that. Also `index` was used as an arg to `plot_coord`, but there are both `x_index` and `y_index` used in the method, so I changed that to `x_index`. I also added a `get_gate_anchor` method since that was needed to clean up the fold issue. All done in https://github.com/Qiskit/qiskit-terra/pull/7545/commits/27abc0ec115b0396c822be26b70060712595efca.

Reply:
Actually, the Anchor code has always been problematic for me, but I left it alone because it worked, But now that it has a problem and the names were being changed, I went ahead and refactored it. There was an unused attribute and every Anchor contained a list of x indexes for some of the other gates in the circuit. I eliminated these and removed 2 methods, so now it is simple and each Anchor only knows the info for 1 gate.